apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V111__setup_claims_availability_function.sql

```diff
@@ -0,0 +1,74 @@
+-- BFD-2549
+-- SETUP_CLAIMS_AVAILABILITY_FUNCTION.SQL
+-- flyway migration to reate a db function that takes a bene_id as an input parameter,
+-- and returns a bit mask denoting available claims data for that beneficiary. This
+-- function will be invoked during ExplanationOfBenefitResourceProvider (STU3, R4) to
+-- improve performance, especially for the case(s) when there are no claims wahtsoever
+-- for a given Claim type for the Beneficiary.
+--
+-- NOTE: HSQL does not support mathematical 'bitwise &' functionality; thus when computing
+--       the bitmask result, we'll simply perform additive functions to derive the same.
+--       PSQL does support support mathematical 'bitwise &' and will therefore be used.
+--
+${logic.hsql-only} CREATE 
+${logic.psql-only} CREATE OR REPLACE 
+FUNCTION check_claims_mask(v_bene_id bigint)
+RETURNS integer
+${logic.psql-only} LANGUAGE plpgsql AS
+${logic.psql-only} $func$
+DECLARE
+	v_rslt           integer  := 0;
+	v_cnt            integer  := 0;         
+                                       -- Java definitions
+	V_CARRIER        integer  := 1;    -- public static final int V_CARRIER_HAS_DATA     = (1 << 0);
+	V_INPATIENT      integer  := 2;    -- public static final int V_INPATIENT_HAS_DATA   = (1 << 1);
+	V_OUTPATIENT     integer  := 4;    -- public static final int V_OUTPATIENT_HAS_DATA  = (1 << 2);
+	V_SNF            integer  := 8;    -- public static final int V_SNF_HAS_DATA         = (1 << 3);
+	V_DME            integer  := 16;   -- public static final int V_DME_HAS_DATA         = (1 << 4);
+	V_HHA            integer  := 32;   -- public static final int V_HHA_HAS_DATA         = (1 << 5);
+	V_HOSPICE        integer  := 64;   -- public static final int V_HOSPICE_HAS_DATA     = (1 << 6);
+	V_PART_D         integer  := 128;  -- public static final int V_PART_D_HAS_DATA      = (1 << 7);
+BEGIN
+	SELECT COUNT(BENE_ID) INTO v_cnt FROM carrier_claims WHERE bene_id = v_bene_id limit 1;
+	IF v_cnt > 0 THEN
+		v_rslt = V_CARRIER;
+	END IF;
+	SELECT COUNT(BENE_ID) INTO v_cnt  FROM inpatient_claims WHERE bene_id = v_bene_id limit 1;
+	IF v_cnt > 0 THEN
+${logic.hsql-only} v_rslt = v_rslt + V_INPATIENT;
+${logic.psql-only} v_rslt = v_rslt # V_INPATIENT;
```

Comment:
I could be missing something, but why are we reassigning `v_rslt` to itself for each of these psql conditional statements? Could we do `v_rslt = v_rslt + V_[CLAIM-TYPE]` for both hsql and psql, then at the end cast the final `v_rslt` to a `::bit`? Would eliminate the psql/hsql only logic.

Reply:
So the using additive ops to emulate a mathematiclal XOR is a bit of a hack; in fact, I had specifically put a comment in the script noting that. While I still prefer the psql support for XOR vs the 'hack', since we only have 8 single-bit masks to worry about, I've just used the additive ops to generate the mask result.

Reply:
Maybe I just don't understand properly how `${logic.psql-only}` works - it just looks like it's only the `hsql` doing the additive ops and `psql` isn't. Is the `hsql-only` logic applying to both ?

Reply:
NVM, my original comment no longer applies to your updated code.